workload.py

```diff
@@ -62,6 +64,42 @@ def generate(self) -> List[Request]:
 
         return ret
     
+class VaryLengthWorkloadGenerator(WorkloadGenerator):
+    """
+    Generate vary length requests from the same context.
+    """
+    def __init__(self, config: WorkloadConfig):
+        super().__init__(config)
+        self.dummy_context = "This is some dummy text. "
+        self.estimated_num_tokens_context = utils.estimate_num_tokens(self.dummy_context)
+        dummy_question = "Index 0. Question: How are you doing today?"
+        self.estimated_num_tokens_question = utils.estimate_num_tokens(dummy_question)
+        self.index = 0
+
+    def generate_context(self) -> str:
+        self.index += 1
+        # The context length pattern: [a 2a 2a 3a 3a 4a 4a ...]
+        return self.dummy_context * (self.config.context_length * ((self.index // 2) + 1) // self.estimated_num_tokens_context)
```

Comment:
Do you think we need an "upper bound" for the context length? The goal is to avoid generate the context longer than the model's capability,